src/Gren/Kernel/Scheduler.js

```diff
@@ -49,6 +49,47 @@ function _Scheduler_receive(callback) {
   };
 }
 
+function _Scheduler_concurrent(tasks) {
+  if (tasks.length === 0) return _Scheduler_succeed([]);
+
+  return _Scheduler_binding(function (callback) {
+    let count = 0;
+    let results = [];
+    let procs = [];
+
+    function killAll() {
+      procs.forEach(_Scheduler_rawKill);
+    }
+
+    function onError(e) {
+      killAll();
+      callback(_Scheduler_fail(e));
+    }
+
+    const batch = tasks.map((task, i) => {
+      function onSuccess(res) {
+        results[i] = res;
+        count++;
+        if (count === tasks.length) {
+          callback(_Scheduler_succeed(results));
+        }
+      }
+      const success = A2(_Scheduler_andThen, onSuccess, task);
+      const handled = A2(_Scheduler_onError, onError, success);
+      return handled;
+    });
+    procs = batch.map(_Scheduler_rawSpawn);
```

Comment:
nitpick: maybe perform the `rawSpawn` inside of `tasks.map` to save one iteration through the array of tasks?

Reply:
There was an edge case I found with this in the experiments repo I can't quite remember (something where the behaviour with an immediate failure was different) - will dig it up and write a test for it.

Reply:
So this edge case was just for the experiments repo, have added tests and refactored to this now :)